testapps/php56_e2e/web/grpc_pubsub.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Google\Auth\ApplicationDefaultCredentials;
+
+function updateAuthMetadataCallback($context)
+{
+    $auth_credentials = ApplicationDefaultCredentials::getCredentials(
+        ['https://www.googleapis.com/auth/pubsub']
+    );
+    return $auth_credentials->updateMetadata($metadata = [],
+                                             $context->service_url);
+}
+
+$channel_credentials = Grpc\ChannelCredentials::createSsl(
+    file_get_contents(__DIR__ . '/../vendor/grpc/grpc/etc/roots.pem'));
+
+$opts = ['credentials' => $channel_credentials,
+         'grpc.ssl_target_name_override' => 'pubsub.googleapis.com'];
+
+$client = new google\pubsub\v1\PublisherClient('pubsub.googleapis.com', $opts);
+
+$deadline = Grpc\Timeval::InfFuture();
+
+$req = new google\pubsub\v1\ListTopicsRequest();
+$proj = getenv('GAE_LONG_APP_ID');
+$req->setProject('projects/' . $proj);
+$call = $client->ListTopics(
+    $req,
+    [],
+    ['call_credentials_callback' => 'updateAuthMetadataCallback']);
```

Comment:
It would be nice if we could use the credentials' [`getUpdateMetadataFunc`](https://github.com/google/google-auth-library-php/blob/master/src/CredentialsLoader.php#L140) function for this:

```php
$auth_credentials = ApplicationDefaultCredentials::getCredentials(
     ['https://www.googleapis.com/auth/pubsub']
);
$call = $client->ListTopics(
     $req,
     [],
     ['call_credentials_callback' => $auth_credentials->getUpdateMetadataFunc())];
```

But it looks like since the first argument doesn't match up, this won't work. Maybe we should change the `google/auth` library to handle this? I assume that is what this function is meant for.